cmd/mesh/upgrade.go

```diff
@@ -157,21 +160,43 @@ func upgrade(rootArgs *rootArgs, args *upgradeArgs, l *logger) (err error) {
 			"error: %v", err)
 	}
 
-	// Check if the upgrade currentVer -> targetVersion is supported
-	err = checkSupportedVersions(currentVer, targetVersion, args.versionsURI, l)
-	if err != nil {
+	// Create a new values translator
+	tr, err := translate.NewReverseTranslator(opversion.OperatorBinaryVersion.MinorVersion)
+	if err != nil && !args.force {
+		return fmt.Errorf("fail to create values translator, error: %v", err)
+	}
+
+	// Translate the current values into ICPS
+	var currentICPS *v1alpha2.IstioControlPlaneSpec
+	if tr != nil {
+		currentICPS, err = tr.TranslateFromValueToSpec([]byte(currentValues))
```

Comment:
I don't think we should be doing this inside of upgrade. It's better to use manifest migrate and then run upgrade on the resulting ICP.

Reply:
talked offline, we will pass in the only icps for both parameters. We will consider API changes maybe in 1.5 or later.